### PR TITLE
[BUG] Medoids empty cluster bug fix

### LINKED ISSUE #752: [BUG] k-medoids produces empty cluster
### Describe the bug

I am trying to perform k-medoids clustering on a dataset containing 50 time series for two observational units. When doing some extreme value testing, I tried to cluster for 10 clusters and got 9 clusters with time series in them, and one empty cluster. Curiously, it's cluster no.7 (not no.0 or no.9, which would have maybe made some intuitive sense). There is no error message.

If I run the clustering for 9 clusters, there is no empty cluster. 

I checked the issues for something related, maybe #672 is somehow relevant?

Figure for illustration:
empty cluster

### Steps/Code to reproduce the bug

I have attached a Jupyter notebook and pickle file to reproduce the bug. Please let me know if you need the data in a different format.

[aeon empty cluster bug.zip](https://github.com/aeon-toolkit/aeon/files/12671238/aeon.empty.cluster.bug.zip)

### Expected results

I expect every cluster to include time series from the data set.

### Actual results

One cluster contains no time series.

### Versions

System:
    python: 3.11.0 (main, Oct 24 2022, 18:26:48) [MSC v.1933 64 bit (AMD64)]
executable: c:\...\python.exe
   machine: Windows-10-10.0.19045-SP0

Python dependencies:
          pip: 22.3
   setuptools: 65.5.0
      sklearn: None
         aeon: 0.3.0
  statsmodels: 0.14.0
        numpy: 1.24.3
        scipy: 1.11.1
       pandas: 2.0.3
   matplotlib: 3.7.2
       joblib: 1.3.1
        numba: 0.57.1
     pmdarima: None
      tsfresh: None

### PR DESCRIPTION
**Pull request by chrisholder**
#### Reference Issues/PRs

Fixes #752 #672 

#### What does this implement/fix? Explain your changes.

There was a bug with PAM implementation where it was performing the wrong swap. I fixed this code and updated the tests.

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/aeon-toolkit/aeon/blob/main/.all-contributorsrc).
- [x] Optionally, I've updated aeon's [CODEOWNERS](https://github.com/aeon-toolkit/aeon/blob/main/CODEOWNERS) to receive notifications about future changes to these files.
- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG] indicating whether the PR topic is related to enhancement, maintenance, documentation, or bug.
- [x] For new estimators, I've added the estimator to the online documentation.

### CODE REVIEW

File: aeon/clustering/clara.py [hunk 1]
```diff
@@ -28,15 +27,17 @@ class TimeSeriesCLARA(BaseClusterer):
     n_clusters : int, default=8
         The number of clusters to form as well as the number of
         centroids to generate.
-    init_algorithm : str, default='random'
+    init_algorithm : str or np.ndarray, default='random'
```
Comment (TonyBagnall) [inline aeon/clustering/clara.py:30]: the type hint in the constructor still lists this as 
        init_algorithm: Union[str, Callable] = "random",

**Review by TonyBagnall — CHANGES_REQUESTED** [review]
two very minor points, otherwise looks good

File: aeon/clustering/k_medoids.py [hunk 2]
```diff
@@ -260,7 +260,10 @@ def _compute_medoids(self, X: np.ndarray, indexes: np.ndarray):
     def _pam_fit(self, X: np.ndarray):
         old_inertia = np.inf
         n_instances = X.shape[0]
-        medoids_idxs = self._init_algorithm(X)
+
+        medoids_idxs = self._init_algorithm
```
[the remark this answers is not in the document]
Comment (chrisholder) [reply aeon/clustering/k_medoids.py]: Fixed in kmedoids and clarans

File: aeon/clustering/clara.py (hunk 1 above)
Comment (chrisholder) [reply aeon/clustering/clara.py:30]: Fixed in clara and clarans

**Review by TonyBagnall — APPROVED** [review]

File: setup.cfg [hunk 3]
```diff
@@ -10,10 +10,10 @@ multi_line_output = 3
...
 addopts =
     --doctest-modules
-    --durations 20
-    --timeout 600
-    --showlocals
-    --numprocesses auto
+;    --durations 20
+;    --timeout 600
+;    --showlocals
+;    --numprocesses auto
```
Comment (MatthewMiddlehurst) [inline setup.cfg:16]: Don't do this in the future

**Pull request merged**